src/mgmt/tests/srv_drpc_tests.c

```diff
@@ -1504,6 +1506,134 @@ test_drpc_pool_create_invalid_acl(void **state)
 
 }
 
+/*
+ * dRPC Rank test utils
+ */
+static void
+expect_daos_resp_with_der(Drpc__Response *resp, int exp_der)
+{
+	Mgmt__DaosResp	*d_resp = NULL;
+
+	assert_int_equal(resp->status, DRPC__STATUS__SUCCESS);
+	assert_non_null(resp->body.data);
+
+	d_resp = mgmt__daos_resp__unpack(NULL, resp->body.len,
+					 resp->body.data);
+	assert_non_null(d_resp);
+	assert_int_equal(d_resp->status, exp_der);
+
+	mgmt__daos_resp__free_unpacked(d_resp, NULL);
+}
+
+/*
+ * dRPC ping rank tests
+ */
+static void
+pack_ping_rank_req(Mgmt__PingRankReq *req, Drpc__Call *call)
+{
+	size_t	len;
+	uint8_t	*body;
+
+	len = mgmt__ping_rank_req__get_packed_size(req);
+	D_ALLOC(body, len);
+	assert_non_null(body);
+
+	mgmt__ping_rank_req__pack(req, body);
+
+	call->body.data = body;
+	call->body.len = len;
+}
+
+static void
+test_drpc_ping_rank_success(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PingRankReq	pr_req = MGMT__PING_RANK_REQ__INIT;
+
+	pack_ping_rank_req(&pr_req, &call);
+
+	ds_mgmt_drpc_ping_rank(&call, &resp);
+
+	expect_daos_resp_with_der(&resp, 0);
+
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+
+}
+
+/*
+ * dRPC kill rank tests
+ */
+static void
+pack_kill_rank_req(Mgmt__KillRankReq *req, Drpc__Call *call)
+{
+	size_t	len;
+	uint8_t	*body;
+
+	len = mgmt__kill_rank_req__get_packed_size(req);
+	D_ALLOC(body, len);
+	assert_non_null(body);
+
+	mgmt__kill_rank_req__pack(req, body);
+
+	call->body.data = body;
+	call->body.len = len;
+}
+
+static void
+test_drpc_kill_rank_success(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__KillRankReq	kr_req = MGMT__KILL_RANK_REQ__INIT;
+
+	pack_kill_rank_req(&kr_req, &call);
+
+	ds_mgmt_drpc_kill_rank(&call, &resp);
+
+	expect_daos_resp_with_der(&resp, 0);
+
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
+/*
+ * dRPC prep shutdown tests
+ */
+static void
+pack_prep_shutdown_req(Mgmt__PrepShutdownReq *req, Drpc__Call *call)
+{
+	size_t	len;
+	uint8_t	*body;
+
+	len = mgmt__prep_shutdown_req__get_packed_size(req);
+	D_ALLOC(body, len);
+	assert_non_null(body);
+
+	mgmt__prep_shutdown_req__pack(req, body);
+
+	call->body.data = body;
+	call->body.len = len;
+}
+
+static void
+test_drpc_prep_shutdown_success(void **state)
```

Comment:
Adding test cases is good - just wondering if there are ways this could fail, which could also be tested. Right now it looks like the function called inside isn't returning anything (or it's not being checked).

Reply:
this item has been added to https://jira.hpdd.intel.com/browse/DAOS-3413